src/contexts/product-context.js

```diff
@@ -0,0 +1,32 @@
+import { createContext, useContext, useReducer } from "react";
+import { productReducer } from "../reducers/product-reducer";
+
+const ProductContext = createContext();
+
+const ProductProvider = ({children}) => {
+
+    const [state, dispatch] = useReducer(productReducer, {
+        data: [],
+        filters: {
+            sortBy: "",
+            category: {
+                p_eq: false,
+                t_eq: false,
+                t_apparel: false,
```

Comment:
the categories are not readable ,can we have more readable names like `protective_equipmet` instead of  `p_eq`and similarly for other two categories